### DISCUSSION TITLE
How to make category columns resize on zoom?

### DISCUSSION THREAD

**Discussion by Eskimo111** [apexcharts/apexcharts.js#3630] [General]
Currently, the width of the column stay the same when we zoom in (with  x-axis category bar chart), this becomes ugly when there is a large number of columns.
![image](https://user-images.githubusercontent.com/62143480/215974829-5cff4760-99db-4d48-858e-60f55052993a.png)
![image](https://user-images.githubusercontent.com/62143480/215975199-e23cdf6e-f184-4872-aff0-fbf96c2ead68.png)
Is there any way to make it resize on zoom in?

**Comment by LentoMan** [comment]
You can try setting a new columnWidth before zoom occurs using something like this perhaps:

```
chart: {
   events: {
      beforeZoom: function(chartContext, { xaxis }) {
         let totalScale = chartContext.w.globals.categoryLabels.length;
         let newScale = (xaxis.max - xaxis.min);
         chartContext.w.config.plotOptions.bar.columnWidth = (totalScale/newScale) * 60 + "%";
    }
}
```
